sources/newRif/rif_utils.py

```diff
@@ -1,33 +1,91 @@
 import os
 import csv
 import re
+import django
+django.setup()
 from sefaria.model import *
 
 path = os.path.dirname(os.path.abspath("top_level_file.txt"))
 
+def cleanspaces(string):
+    while any(space in string for space in ['  ', '( ', ' )', ' :', ' .']):
+        for space in ['  ', '( ', ' )', ' :', ' .']:
+            string = string.replace(space, space.replace(' ', '', 1))
+    return string.strip()
+
 def removeinbetween(stringtoclean, sub1, sub2):
     return re.sub(sub1+'.*?'+sub2, '', stringtoclean)
 
 def remove_notes(string, masechet):
     for note_tag in [tags_map[masechet][tag] for tag in ['gemara_refs', 'notes', 'tanakh_refs']]:
-        removeinbetween(string, note_tag, map[masechet]['end_tag'])
+        string = removeinbetween(string, note_tag, tags_map[masechet]['end_tag'])
+    return string
 
 def hebrewplus(string_to_clean, letters_to_remain=''):
     return re.sub(r"[^א-ת "+letters_to_remain+"]", '', string_to_clean)
 
+def remove_metadata(string, masechet):
+    string = remove_notes(string, masechet)
+    string = re.sub(r'@00פרק .*? ', ' ', string)
+    string = re.sub(mefarshim_tags(masechet), '', string)
+    string = hebrewplus(string, '"\'')
+    return cleanspaces(string)
+
 def netlen(string): #length in words without oher text
     for tag in '346':
         string = removeinbetween(string, '@1'+tag, '@77')
     return len(hebrewplus(string).split())
 
+def netlen2(string, masechet): #a better version.
+    return len(remove_metadata(string, masechet).split())
+
 def get_hebrew_masechet(masechet):
     return Ref(masechet).index.get_title('he')
 
-def open_rif_file(masechet):
-    for root, dirs, files in os.walk(os.getcwd()+'/rif'):
+def open_rif_file(masechet, path='/rif'):
+    for root, dirs, files in os.walk(os.getcwd()+path):
         for file in files:
-            if get_hebrew_masechet(masechet) in file:
-                return open(root+'/'+file, encoding = 'utf-8')
+            if get_hebrew_masechet(masechet) in file or masechet in file:

```

Comment:
Here as well, let's try and avoid this pattern. Filenames should be derived from first principles rather than scanning the disk for a potential match.